src/sympc/tensor/mpc_tensor.py

```diff
@@ -171,19 +188,48 @@ def generate_przs(
 
     @staticmethod
     def generate_shares(
-        secret: ShareTensor,
+        secret: Union[ShareTensor, torch.Tensor, float, int],
         nr_parties: int,
         tensor_type: Optional[torch.dtype] = None,
+        **kwargs,
     ) -> List[ShareTensor]:
-        """Given a secret, split it into a number of shares such that
-        each party would get one
-
-        :return: list of shares
-        :rtype: list of ShareTensor
+        """Given a secret, split it into a number of shares such that each
+        party would get one.
+
+        Args:
+            secret (Union[ShareTensor, torch.Tensor, float, int]): secret to split
+            nr_parties (int): number of parties to split the scret
+            tensor_type (torch.dtype, optional): tensor type. Defaults to None.
+            **kwargs: keywords arguments passed to ShareTensor
+
+        Returns:
+            List[ShareTensor]. List of ShareTensor
+
+        Examples:
+            >>> from sympc.tensor.mpc_tensor import MPCTensor
+            >>> MPCTensor.generate_shares(secret=2, nr_parties=2)
+            [[ShareTensor]
+                | [FixedPointEncoder]: precision: 16, base: 2
+                | Data: tensor([15511500.]), [ShareTensor]
+                | [FixedPointEncoder]: precision: 16, base: 2
+                | Data: tensor([-15380428.])]
+            >>> MPCTensor.generate_shares(secret=2, nr_parties=2,
+                encoder_base=3, encoder_precision=4)
+            [[ShareTensor]
+                | [FixedPointEncoder]: precision: 4, base: 3
+                | Data: tensor([14933283.]), [ShareTensor]
+                | [FixedPointEncoder]: precision: 4, base: 3
+                | Data: tensor([-14933121.])]
         """
 
-        if not isinstance(secret, ShareTensor):
-            raise ValueError("Secret should be a ShareTensor")
+        if not isinstance(secret, (ShareTensor, torch.Tensor, float, int)):
+            raise ValueError(
+                "Secret should be a ShareTensor, torchTensor, float or int."
+            )
+
+        # if secret is not a ShareTensor, a new instance is created
+        if isinstance(secret, (torch.Tensor, float, int)):
```

Comment:
Q: Can't we have this first and then the other one?
Like:
```
if isinstance(secret, (torch.Tensor, float, int)):
     secret = ShareTensor...

if not isinstance(secret, ShareTensor):  # At this point it should be a share)
   raise...
```

Reply:
Sure!